test/files/nc/postgrey_whitelist.txt

```diff
@@ -6,4 +6,3 @@ client_name=whitelist.tld
 helo_name=whitelist.tld
 sender=test@whitelist.tld
 recipient=user1@localhost.localdomain
-
```

Comment:
> * [This page conveys what a "final newline" actually is](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/missing-final-newline.html), ensuring the last line ends with `\n` / `LF` (_which our `.gitattributes` ensures on commit now_).

Out of curiosity: Do I miss something or are all examples on the linked page identically (with the exception of the comments) and should all produce the same output? 😆

Reply:
> are all examples on the linked page identical
> and should all produce the same output?

The print statements are yes, they're not relevant, line endings are (_which the comments convey_).

Basically:

```
print("Hello")LF
print("world")EOF
```

```
print("Hello")LF
print("world")LF
EOF
```

Where the LF (`\n`) terminates a line. So the first example with `wc -l` should be 2 lines, while the 2nd example should output 3 lines?

A final newline here is just the document ending with `LF` AFAIK, whereas I had the habit of doing so with a blank line (`\n\n`) 😅 

```
print("Hello")LF
print("world")LF
LF
EOF
```

Sometimes in Github review diffs, there was a hint shown about missing final new line, and it was about the lack of the trailing `LF`, rather than a blank line (_which likewise fixes it_).

Reply:
Yeah, I understand the problem that was explained. I was just wondering about their examples, why the did not use examples that actually works like:

``printf '%s\n' Hello #LF`` 
``printf '%s'   Hello #EOF`` 
etc.

Reply:
> I was just wondering about their examples, why the did not use examples that actually works

Oh, probably not to risk adding any confusion to that having meaning to what they're showing?

Perhaps they should have used content that wouldn't print anything out like a dummy function `no_op()`.